apps/api/scripts/generate-metadata.ts

```diff
@@ -0,0 +1,32 @@
+import fs from 'node:fs';
+import path from 'node:path';
+import { PluginMetadataGenerator } from '@nestjs/cli/lib/compiler/plugins';
+import { ReadonlyVisitor } from '@nestjs/swagger/dist/plugin';
+
+const tsconfigPath = 'tsconfig.build.json';
+const srcPath = path.join(__dirname, '..', 'src');
+const metadataPath = path.join(srcPath, 'metadata.ts');
+
+/*
+ * We create an empty metadata file to ensure that files importing `metadata.ts`
+ * will compile successfully before the metadata generation occurs.
+ */
+const defaultContent = `export default async () => { return {}; };`;
+
+fs.writeFileSync(metadataPath, defaultContent, 'utf8');
+console.log('metadata.ts file has been generated with default content.');
+
+/**
+ * This file is responsible for generating Nest.js metadata for the API.
```

Comment:
<!-- Nitpicks are trivial preference-based requests. These should be non-blocking by nature.  -->
🤓 **nitpick (non-blocking):** ‏
```suggestion
 * This code snippet is responsible for generating Nest.js metadata for the API.
```

Reply:
Alternatively, move the entire comment block to the top of the file

Reply:
Good call - I moved it to the top of the file.